io.openems.edge.ess.sinexcel/src/io/openems/edge/ess/sinexcel/EssSinexcel.java

```diff
@@ -219,6 +219,8 @@ private void setBatteryRanges() throws OpenemsNamedException {
 		final double EFFICIENCY_FACTOR = 0.9;
 		this._setAllowedChargePower((int) (chaMaxA * chaMaxV * -1 * EFFICIENCY_FACTOR));
 		this._setAllowedDischargePower((int) (disMaxA * disMinV * EFFICIENCY_FACTOR));
+		
+		this._setCapacity(battery.getCapacity().get());
```

Comment:
Aus meiner Sicht würde das eher in die `initializeBattery` methode passen: https://github.com/OpenEMS/openems/blob/develop/io.openems.edge.ess.sinexcel/src/io/openems/edge/ess/sinexcel/EssSinexcel.java#L153

(Wobei das Vorgehen dort extrem gefährlich ist, weil diese Listener nie entfernt werden. Das kann dazu führen, dass der GarbageCollector deaktivierte Komponenten nicht löschen kann.)